### PR TITLE
LinkedIn OAuth Integration

### PR DESCRIPTION
**Pull request by aidanouckama**
Played around with the repo and Hono, learned a lot while integrating this! Did a lil janky type fix for creating middleware outside of the main `packages\server\src\trpc.ts`, let me know what you guys think.

❗ CHANGES TO `.env` AND `wrangler.toml`

[ NEW ] `packages/server/src/middlewares/auth.ts` - Middleware for auth + authentication for routes
`packages/server/src/trpc.ts` - Added `authenticatedRoute`, a protected route for authenticated users
[ NEW ] `packages/server/src/types/middleware.ts` - Type for encapsulated tRPC middleware functions

`packages/server/package.json` - Added dependency: `@hono/oauth-providers`
`packages/server/src/context.ts` - Added user to context

`packages/ui/*` - Fixed some paths

### CODE REVIEW

File: packages/server/src/middlewares/auth.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      throw new TRPCError({ code: "UNAUTHORIZED", message: "Not authenticated" });
+    }
+  
+    return opts.next({
+      ctx: {
+        ...opts.ctx,
+        user: {
+          ...user,
+          isAdmin: user.email?.endsWith("@knighthacks.org"),
+        },
```
Comment (Lewin-B) [inline packages/server/src/middlewares/auth.ts]: Ayo?

File: packages/server/src/trpc.ts [hunk 2]
```diff
@@ -2,9 +2,19 @@ import { initTRPC } from "@trpc/server";
...
 });
 
 export const router = t.router;
-export const publicProcedure = t.procedure; // unprotected procedure
+
+// unprotected procedure
+export const publicProcedure = t.procedure;
+
+// user is signed-in procedure
+export const authenticatedProcedure = t.procedure.use(isAuthenticated);
```
Comment (Lewin-B) [inline packages/server/src/trpc.ts:19]: Nice.

**Review by Lewin-B — CHANGES_REQUESTED** [review]
Just a quick change with the isAdmin

**Review by Lewin-B — APPROVED** [review]
lgtm

File: packages/server/src/middlewares/auth.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  if (!user) {
+    throw new TRPCError({ code: "UNAUTHORIZED", message: "Not authenticated" });
+  }
+
+  return next({
+    ctx: {
+      ...ctx,
+      user: {
+        ...user,
+        // isAdmin: user.email?.endsWith("@pipelines.lol"), - Possible way to have admins
```
Comment (Nyumat) [inline packages/server/src/middlewares/auth.ts:33]: Neat.

**Review by Nyumat — APPROVED** [review]

File: packages/server/src/middlewares/auth.ts (hunk 1 above)
Comment (Nyumat) [reply packages/server/src/middlewares/auth.ts]: 🤣

**Pull request merged**